### DISCUSSION TITLE
[FEATURE]固定 docker 容器 ip

### DISCUSSION THREAD

**Discussion by Fantasy135** [1Panel-dev/1Panel#6344] [Ideas]
### 1Panel 版本

1.10.14tls

### 请描述您的需求或者改进建议

建议应用在启动时指定 ip，这样就不会在服务器重启之后，由于启动顺序导致 ip 变化

### 请描述你建议的实现方案

在安装应用时，如果使用的是 1panel-network 网卡，根据已经使用的 ip 往后延顺
并在生产的 docker-compose 配置文件 添加上 指定 ip 的方法
 networks:
      1panel-network:
        ipv4_address: 根据已经使用的 ip 往后延顺的

这样可以解决由于重启系统，或 docker 原因，由于容器的启动顺序而导致反代服务器由于反代 ip 的服务变化，而造成的问题

**Comment by see-more** [comment]
容器通讯用服务名就行了吧

**Reply by chenxuuu** [reply]
openresty反代那里是不允许直接填容器名+端口的

**Reply by zhengkunwang223** [reply]
> openresty反代那里是不允许直接填容器名+端口的

直接用 127.0.0.1:容器外部端口

**Comment by wanghe-fit2cloud** [comment]
> 容器通讯用服务名就行了吧

是的，使用服务名或者容器名是可以的。
如果必须使用 IP，也可以手动编辑 compose 文件来实现。
如果是在容器页面手动创建的容器，也支持设置容器 IP。

**Reply by Fantasy135** [reply]
我是需要 openresty 那里反代出去， 但是 openresty 那边一键部署，并不完整啊，已安装应用我这边是空白，新安装应用也只是个别可选
